src/modules/roc_sndio/target_sox/roc_sndio/sox_source.cpp

```diff
@@ -321,6 +322,7 @@ bool SoxSource::open_() {
     }
 
     is_file_ = !(input_->handler.flags & SOX_FILE_DEVICE);
+    sample_spec_.set_sample_rate((unsigned long)input_->signal.rate);
```

Comment:
Could you please move error checking below (`if (input_->signal.channels != n_channels_) {`) before these lines?